spharpy/spherical.py

```diff
@@ -1005,3 +1007,413 @@ def sph_identity_matrix(n_max, type='n-nm'):
         identity_matrix[n, linear_nm] = 1
 
     return identity_matrix
+
+
+
+class SphericalHarmonics:
+    r"""
+    Compute spherical harmonic basis matrices, their inverses, and gradients.
+    The spherical harmonic Ynm is given by:
+    .. math::
+        Y_{nm} = N_{nm} P_{nm}(cos(\theta)) T_{nm}(\phi)
+
+    where:
+    - :math:`n` is the degree
+    - :math:`m` is the order
+    - :math:`P_{nm}` is the associated Legendre function
+    - :math:`N_{nm}` is the normalization term
+    - :math:`T_{nm}` is a term that depends on whether the
+     harmonics are real or complex
+    - :math:`\theta` is the colatitude (angle from the positive z-axis)
+    - :math:`\phi` is the azimuth (angle in the x-y plane from the x-axis)
+
+    The normalization term :math:`N_{nm}` is given by:
+
+    .. math::
+        N_{nm}^{\text{SN3D}} =
+        \sqrt{\frac{2n+1}{4\pi} \frac{(n-|m|)!}{(n+|m|)!}}
+
+        N_{nm}^{\text{N3D}} = N_{nm}^{\text{SN3D}} \sqrt{\frac{2n+1}{2}}
+
+        N_{nm}^{\text{MaxN}} = ... (max of N3D)
+
+    The associated Legendre function :math:`P_{nm}` is defined as:
+
+    .. math::
+        P_{nm}(x) = (1-x^2)^{|m|/2} (d/dx)^n (x^2-1)^n
+
+    The term :math:`T_{nm}` is defined as:
+        - For complex-valued harmonics:
+            .. math::
+                T_{nm} = e^{im\phi}
+        - For real-valued harmonics:
+            .. math::
+                T_{nm} = \begin{cases}
+                            \cos(m\phi) & \text{if } m \ge 0 \\
+                            \sin(m\phi) & \text{if } m < 0
+                        \end{cases}
+
+    The spherical harmonics are orthogonal on the unit sphere, i.e.,
+
+    .. math::
+        \int_{sphere} Y_{nm} Y_{n'm'}* d\omega = \delta_{nn'} \delta_{mm'}
+
+    where:
+    - :math:`*` denotes complex conjugation
+    - :math:`\delta_{nn'}` is the Kronecker delta function
+    - :math:`d\omega` is the solid angle element
+    - The integral is over the entire sphere
+
+    The class supports the following conventions:
+
+    - normalization: Defines the normalization convention.
+     It can be 'n3d', 'maxN', or 'sn3d'.
+        - ``'n3d'``: Uses the 3D normalization
+        (also known as Schmidt semi-normalized).
+        - ``'maxN'``: Uses the maximum norm
+        (also known as fully normalized).
+        - ``'sn3d'``: Uses the SN3D normalization
+        (also known as Schmidt normalized).
```

Comment:
```suggestion
    - basis_type: Defines the type of spherical harmonic basis.

      - ``'complex'``: Uses complex-valued spherical harmonics.
      - ``'real'``: Uses real-valued spherical harmonics.

    - ``'normalization'``: Defines the normalization convention:

      - ``'n3d'``: Uses the 3D normalization
        (also known as Schmidt semi-normalized).
      - ``'maxN'``: Uses the maximum norm
        (also known as fully normalized).
      - ``'sn3d'``: Uses the SN3D normalization
        (also known as Schmidt normalized).
```
this fixes the intend, was it intended to remove ``'basis_type'``?

Reply:
yes, because this description is already covered by the initialisation (parameter) doc. So it's like a duplicate here.